### PR TITLE
Oi: Version 4.000 added



### LINKED ISSUE #8075: Update Oi with Arabic
Following #3820 oi now has Arabic support and ssa latin.

### PR DESCRIPTION
**Pull request by yanone**
Taken from the upstream repo https://github.com/kosbarts/Oi at commit https://github.com/kosbarts/Oi/commit/1938199013c7810522c354d2a885401e6a319ab1.

Resolves #8075
## PR Checklist:

- [ ] `minisite_url` definition in the METADATA.pb file for commissioned projects
- [ ] `primary_script` definition in the METADATA.pb file for all projects that have a primary non-Latin based language support target
- [ ] `subsets` definitions in the METADATA.pb reflect the actual subsets and languages present in the font files (in alphabetic order). For **CJK fonts**, only include one of the following subsets `chinese-hongkong`, `chinese-simplified`, `chinese-traditional`, `korean`, `japanese`.
- [x] Fontbakery checks are reviewed and failing checks are resolved in collaboration with the upstream font development team
- [x] Diffenator2 regression checks for revisions on all projects that are currently in production
- [ ] Designers bio info have to be present in the designer catalog (at least an issue should be opened for tracking this, if they are not)
- [ ] Check designers order in metadata.pb, since the first one of the list appears as “principal designer”
- [ ] Social media formatted visual assets for all new commissioned projects in the Drive directory, communicate with the repository Maintainer so that they can push this content to the Social Media tracker spreadsheet
- [ ] Social media content draft for all new commissioned projects in the Drive directory and Social Media tracker spreadsheet, communicate with the repository Maintainer so that they can push this content to the Social Media tracker spreadsheet

### CODE REVIEW

**Comment by yanone** [comment]
Ignore all Arabic mark attachment fails. I manually confirmed all of these to work. I don't understand why these reports are generated here.

Also ignore the report about ALEF MAKSURA, as the letter only exists in final and isolated form. This is incorrectly reported by `youseedee`; I'm investigating. Most likely a mistake in Unicode.

The `saria heh` as well as the missing African glyphs: This depends on when the font update was commissioned. The African is constantly improved, and the `saria heh` check in `shaperglot` was added 7 months ago, likely after the update was commissioned. I can't give a recommendation on whether or not these should be fixed.

**Comment by yanone** [comment]
Also ignore all Eng fails. This uses the African Eng as default but shaperglot expects the European Eng as default, which needs to be defined in/for shaperglot.

File: ofl/oi/METADATA.pb [hunk 1]
```diff
@@ -22,6 +22,7 @@ subsets: "tamil"
 subsets: "vietnamese"
```
Comment (davelab6) [inline ofl/oi/METADATA.pb:22]: no subset "arabic" added..?

**Comment by yanone** [comment]
Update: The ALEF MAKSURA report is accurate, not a mistake, but it doesn't matter in this case. The letter is used in intial and medial form in Uyghur for example, but not in Arabic/Persian/Urdu, which this font here is targetting.

We're gonna have to get used to this report for now, as we currently have no way to separate this per glyphset because only one unicode is involved but two presentation forms of it are unnecessary (and uncommon in font files) for the major Arabic script orthographies.

**Comment by yanone** [comment]
Also ignore the `saria heh` shaping FAIL. Simon mentioned that the respective issue https://github.com/notofonts/arabic/issues/230 is still unsolved, so we should ignore it for now.

**Comment by emmamarichal** [comment]
Hi @yanone!

I'm not expert in Arabic, so I'm afraid I can't give a precise review for this. However, I noticed this, that looked weird to me:

Screenshot 2024-10-25 at 11 00 59

Screenshot 2024-10-25 at 11 06 02

It concerns `/uni063A.medi/uni0641.fina/uni0641.medi/uni06A4.fina/uni06A4.medi/uni06A1.fina/uni06A1.medi/uni06A7.medi/uni06A8.medi`

**Comment by yanone** [comment]
That's not specific to Arabic, but an outline issue. I'll look into it, thank you.

**Comment by yanone** [comment]
Updated

**Review by emmamarichal — APPROVED** [review]

**Pull request merged**